src/components/ui/Navigation.vue

```diff
@@ -35,26 +35,40 @@ import { getModule } from 'vuex-module-decorators'
 import FlashesModule from '../../modules/FlashesModule'
 import Flash from '../../models/Flash'
 import { FlashType } from '../../models/Flash'
+import Web3Module from '../../modules/Web3Module'
+import Listing from '../../models/protocol/Listing'
 
 import '@/assets/style/ui/navigation.sass'
+import ContractsAddresses from '../../models/ContractAddresses'
 
 @Component
 export default class Navigation extends Vue {
-  protected async setAddress(e: Event) {
+  protected async setPublicWalletAddress(e: Event) {
+
     e.preventDefault()
     e.stopPropagation()
+
     const flashesModule = getModule(FlashesModule, this.$store)
     const metaMaskModule = getModule(MetaMaskModule, this.$store)
+    const web3Module = getModule(Web3Module, this.$store)
 
     const result = await enable()
-    if (typeof result !== 'string') {
-      const flash = new Flash(Errors.METAMASK_NOT_CONNECTED, FlashType.error)
-      flashesModule.append(flash)
-    } else {
-      metaMaskModule.setAddress(result)
-      const flash = new Flash(Messages.METAMASK_CONNECTED, FlashType.info)
-      flashesModule.append(flash)
+    const accept = typeof result === 'string'
+
+    let message = Errors.METAMASK_NOT_CONNECTED
+    let flashType = FlashType.error
+
+    switch (accept) {
```

Comment:
switch should not be used for a binary (true/false). Use an `if` here

or use a ternary in your let's above: `const message = accept ? Messages.Foo : Errors.Bar`

Reply:
So this is something I've been playing with trying to use a more pattern-matching approach.  It does seem heavy handed for simple ifs.